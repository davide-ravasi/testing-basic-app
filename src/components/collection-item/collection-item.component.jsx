import React from 'react';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cart.actions';

import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer
} from './collection-item.styles';

export const CollectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className='image' imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer className="item-name">{name}</NameContainer>
        <PriceContainer className="item-price">{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButton className="item-add" onClick={() => addItem(item)} inverted>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
};

export const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(
  null,
  mapDispatchToProps
)(CollectionItem);
