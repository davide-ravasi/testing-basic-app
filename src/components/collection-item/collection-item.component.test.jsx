import React from 'react';
import { shallow } from 'enzyme';
import { CollectionItem, mapDispatchToProps } from './collection-item.component';

describe('Collection-item component', () => {
    let wrapper, mockProps;

    beforeEach(() => {
        mockProps = {
            item: {
                imageUrl: 'https://i.ibb.co/bPmVXyP/black-converse.png',
                name: "Black Converse",
                price: 110,
                quantity: 1
            },
            addItem: jest.fn()
        }

        wrapper = shallow(<CollectionItem {...mockProps} />);
    })

    it('render collection item component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('display the correct name', () => {
        const btn = wrapper.find('.item-name');

        expect(btn.text()).toMatch(mockProps.item.name);
    })

    it('display the correct price', () => {
        const btn = wrapper.find('.item-price');

        expect(btn.text()).toEqual(mockProps.item.price.toString());
    })

    it('display the correct imageURl', () => {
        const image = wrapper.find('.image');

        expect(image.props()["imageUrl"]).toBe(mockProps.item.imageUrl);
    });

    it('call  addItem action', () => {
        wrapper.find('.item-add').simulate('click');

        expect(mockProps.addItem).toHaveBeenCalled();
    })

    it('call dispatch when  addItem was called', () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);

        props.addItem();

        expect(dispatch).toHaveBeenCalled();
    })

})