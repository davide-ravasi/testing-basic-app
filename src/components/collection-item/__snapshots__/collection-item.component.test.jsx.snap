// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection-item component render collection item component 1`] = `
<styled.div>
  <styled.div
    className="image"
    imageUrl="https://i.ibb.co/bPmVXyP/black-converse.png"
  />
  <styled.div>
    <styled.span
      className="item-name"
    >
      Black Converse
    </styled.span>
    <styled.span
      className="item-price"
    >
      110
    </styled.span>
  </styled.div>
  <Styled(CustomButton)
    className="item-add"
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </Styled(CustomButton)>
</styled.div>
`;
